amandas-project/src/containers/Game.js

```diff
@@ -0,0 +1,160 @@
+import React from 'react'
+import Cell from '../components/Cell'
+
+const CELL_SIZE = 20
+const WIDTH = 400
+const HEIGHT = 400
+
+class Game extends React.Component {
+  constructor() {
+    super()
+    this.rows = HEIGHT / CELL_SIZE
+    this.cols = WIDTH / CELL_SIZE
+    this.board = this.makeEmptyBoard()
+  }
+
+  state = {
+    cells: [],
+    interval: 100,
+    isRunning: false
+  }
+
+  runGame = () => { // runs the game 
+    this.setState({
+      isRunning: true
+    })
+    this.runIteration()
+  }
+
+  stopGame = () => { // stops the game
+    this.setState({
+      isRunning: false
+    })
+    if(this.timeoutHandler) {
+      window.clearTimeout(this.timeoutHandler)
+      this.timeoutHandler = null
+    }
+  }
+
+  runIteration() { // runs the game based on rules and a time interval
+    let newBoard = this.makeEmptyBoard()
+
+    for(let y = 0; y < this.rows; y++) {
+      for(let x = 0; x < this.cols; x++) {
+        let neighbors = this.calculateNeighbors(this.board, x, y)
+        if(this.board[y][x]) {
+          if(neighbors === 2 || neighbors === 3) {
+            newBoard[y][x] = true
+          } else {
+            newBoard[y][x] = false
+          }
+        } else {
+          if(!this.board[y][x] && neighbors === 3) {
+            newBoard[y][x] = true
+          }
+        }
+      }
+    }
```

Comment:
I really would like you to have all of this game logic code on a separate file and not here inside the game component, this makes your code very difficult to understand